lib/modules/hosts/youtube.js

```diff
@@ -5,7 +5,7 @@ export default {
 	name: 'youtube',
 	attribution: false,
 	domains: ['youtube.com', 'youtu.be'],
-	detect: ({ href }) => (/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href),
+	detect: ({ href }) => (/^https?:\/\/(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/?(?:watch|embed)(?:.*?v=|v\/|\/))([\w-]+)/i).exec(href),
```

Comment:
@erikdesjardins  :+1: Squashed and commited. Only change I did was the array into the regex. The regex is slightly faster, and case-insensitive.